### ISSUE TITLE
Issue with media controls on Tivo Bolt DVR

### ISSUE THREAD

**Issue by addd45** [groupwhere/homeassistant-tivo#4]
So basically none of my media functions work 🤣, so I'll just post some relevant logs for each:
This is a Tivo Bolt running the non latest (buggy) software update. Connected via ethernet. I tested during a recording and vise-versa. The tivo was online and during live tv.

Play/Pause:
```
Error executing service <ServiceCall media_player.media_play_pause: entity_id=['media_player.tivo_dvr']>

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/homeassistant/core.py", line 1002, in _event_to_service_call
    await service_handler.func(service_call)
  File "/usr/lib/python3.6/site-packages/homeassistant/components/media_player/__init__.py", line 403, in async_service_handler
    yield from getattr(player, method['method'])(**params)
  File "/usr/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
TypeError: 'NoneType' object is not callable
```
Next: (relevant traceback)
```
  File "/config/custom_components/media_player/tivo.py", line 480, in media_next_track
    self.media_ch_up()
  File "/config/custom_components/media_player/tivo.py", line 324, in media_ch_up
    data = self.send_code('CHANNELUP')
  File "/config/custom_components/media_player/tivo.py", line 238, in send_code
    data = self.sock.recv(bufsize)
ConnectionResetError: [Errno 104] Connection reset by peer
```

Previous: (relevant traceback)

```
  File "/config/custom_components/media_player/tivo.py", line 468, in media_previous_track
    self.media_ch_dn()
  File "/config/custom_components/media_player/tivo.py", line 337, in media_ch_dn
    self._current["status"]  = words[2]
IndexError: list index out of range
```

Obviously I can provide more detail if needed, just let me know what info.

**Comment by groupwhere** [comment]
It sort of looks like it's not able to connect at all.  Can you try setting debug: 1 in your configuration.yaml?  Also you may be able to test yourself by running "telnet ip.ad.re.ss 31339" from your linux machine, or even windows.  telnet would need to be installed in both cases.

**Comment by addd45** [comment]
Yeah it looks like I couldn't connect via telnet. Doesn't make sense though because it had to have connected at one point, but not sure how/when.

FWIW I connect via the tivo app and can connect via http?

**Comment by groupwhere** [comment]
That uses a different protocol altogether that I would like to get going as well.  It would be a different component, most likely.  Try checking that the Tivo is in livetv mode first and/or restart it.  Telnet should definitely work.

**Comment by addd45** [comment]
Ok, I'll try that when I get home

**Comment by addd45** [comment]
So when I connect via putty I get "COMMAND_TIMEOUT" upon connecting. typing anything gets me "INVALID_COMMAND"

ie I think I'm at least connecting, but I'm not sure how to test any functionality. 

update:

connecting via my linux terminal connects fine and then closes immediately. I restarted the Bolt with no success.

**Comment by groupwhere** [comment]
Is your Tivo network connected via moca, ethernet or wifi?  Just wondering if you have a network issue.  COMMAND TIMEOUT is normal if you wait more than a few seconds - annoying, I know.  Otherwise, the syntax must be exact, and I believe in CAPS.  See the pdf linked via the README.md.

**Comment by addd45** [comment]
The whole entire network is connected via ethernet. 

EDIT: Okay looks like my syntax was wrong on windows lol. So I can connect from CMD and Putty, but linux was finicky for some reason. Gonna test out some commands now I guess

**Comment by addd45** [comment]
FYI here's the debug logs. Pretty much repeats this over and over again

![image](https://user-images.githubusercontent.com/33461111/40202963-4a93c642-59e9-11e8-947f-7f5ca8a69f62.png)

**Comment by groupwhere** [comment]
Is there any firewall running on the homeassistant machine?  If you can connect from that particular machine via telnet then it should work.  Please verify that the port is 31339 and not 3l339 or something like that in your config.

**Comment by addd45** [comment]
Yeah the port is correct and unless Hassio has a firewall I'm unaware of, it can't be that.. Please note that this isn't a 100% unable to connect scenario, so I tend to believe this isn't an issue with my setup. 

Attached is a photo of an actual (weird) response being received, but still failing 
![image](https://user-images.githubusercontent.com/33461111/40754456-1ef4fec4-643f-11e8-8148-5ababd1d79ce.png)


I'm starting to think maybe the Tivo Bolt is more finicky with the connections. How does the code handle keeping the connection alive / avoiding timeouts?

**Comment by groupwhere** [comment]
It connects as needed and then closes the socket when it's done.  This is mainly because the Tivo will send a timeout response if it doesn't get any input on the connection.  I used the Bolt during development but it is a 2 year old original Bolt, fwiw.

**Comment by groupwhere** [comment]
Also, I was reminded today, the plugin will not be able to talk to tivo if you are in a menu or perhaps even the guide.  This is one of the hazards of using the telnet interface.

**Comment by jcollas** [comment]
The play/pause functions and the exception were fixed in 0.3.0
